src/auth/rest-current-user.ts

```diff
@@ -0,0 +1,14 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+export interface AuthUser {
+  sub: string;
+  permissions?: string[];
+}
+
+export const RestCurrentUser = createParamDecorator(
+  (data: unknown, context: ExecutionContext): AuthUser => {
+    const request = context.switchToHttp().getRequest();
+
+    return request.user;
+  },
+);
```

Comment:
Very useful this decorator! Maybe we can rename to only current-user.ts and remove prefix rest-